tools/core2js.rkt

```diff
@@ -0,0 +1,186 @@
+#lang racket
+
+(require "common.rkt" "fpcore.rkt")
+(provide compile-program)
+
+(define (fix-name name)
+  (string-join
+    (for/list ([char (~a name)])
+      (if (regexp-match #rx"[a-zA-Z0-9_]" (string char))
+          (string char)
+          (format "_~a_" (char->integer char))))
+    ""))
+
+(define/match (constant->js c)
+  [('E) "math.E"]
+  [('LOG2E) "math.LOG2E"]
+  [('LOG10E) "math.LOG10E"]
+  [('LN2) "math.LN2"]
+  [('LN10) "math.LN10"]
+  [('PI) "math.pi"]
+  [('PI_2) "(math.pi/2)"]
+  [('PI_4) "(math.pi/4)"]
+  [('1_PI) "(1/math.pi)"]
+  [('2_PI) "(2/math.pi)"]
+  [('2_SQRTPI) "(2/math.sqrt(math.pi))"]
+  [('SQRT2) "math.SQRT2"]
+  [('SQRT1_2) "(1/math.SQRT2)"]
+  [('MAXFLOAT) "Number.MAX_VALUE"]
+  [('TRUE) "true"]
+  [('FALSE) "false"]
+  [('INFINITY) "Infinity"]
+  [('NAN) "math.NaN"]
+  [(_) (error 'constant->js "Unsupported constant ~a" c)])
+
+(define/match (operator->js op)
+  [((or '== '+ '- '* '/  '< '> '<= '>=)) (format "(~a ~a ~a)" "~a" op "~a")]
+  [('fabs) "math.abs(~a)"]
+  [('exp) "math.exp(~a)"]
+  [('exp2) "math.pow(2, ~a)"]
+  [('expm1) "math.expm1(~a)"]
+  [('log) "math.log(~a)"]
+  [('log10) "math.log10(~a)"]
+  [('log2) "math.log2(~a)"]
+  [('log1p) "math.log1p(~a)"]
+  [('logb) "math.floor(math.log2(math.abs(~a)))"]
+  [('pow) "math.pow(~a, ~a)"]
+  [('sqrt) "math.sqrt(~a)"]
+  [('cbrt) "math.cbrt(~a)"]
+  [('hypot) "math.hypot(~a, ~a)"]
+  [('sin) "math.sin(~a)"]
+  [('cos) "math.cos(~a)"]
+  [('tan) "math.tan(~a)"]
+  [('asin) "math.asin(~a)"]
+  [('acos) "math.acos(~a)"]
+  [('atan) "math.atan(~a)"]
+  [('atan2) "math.atan2(~a, ~a)"]
+  [('sinh) "math.sinh(~a)"]
+  [('cosh) "math.cosh(~a)"]
+  [('tanh) "math.tanh(~a)"]
+  [('asinh) "math.asinh(~a)"]
+  [('acosh) "math.acosh(~a)"]
+  [('atanh) "math.atanh(~a)"]
+  [('erf) "math.erf(~a)"]
+  [('erfc) "1 - math.erf(~a)"]
```

Comment:
This isn't a good way to implement this, since for large `a` this will have a lot of rounding error. We might not have a better implementation, I guess, but it's going to have a lot of error.

Reply:
Should this be removed or just left as-is? There is also https://www.npmjs.com/package/math-erfc but I'm not sure it's good to rely on too many external packages.

Reply:
I think we should leave it but leave a TODO comment. You could possibly open a bug report in math.js asking for support.